flink/v1.20/flink-runtime/NOTICE

```diff
@@ -63,29 +63,277 @@ NOTICE file:
 
 --------------------------------------------------------------------------------
 
-This binary artifact includes Apache Yetus with the following in its NOTICE
+This binary artifact includes Project Nessie with the following in its NOTICE
 file:
 
-| Apache Yetus
-| Copyright 2008-2020 The Apache Software Foundation
-|
+| Dremio
+| Copyright 2015-2017 Dremio Corporation
+| 
+| This product includes software developed at 
+| The Apache Software Foundation (http://www.apache.org/).
+| 
+| Apache Avro
+| Copyright The Apache Software Foundation
+| 
+| Apache Commons
+| Copyright The Apache Software Foundation
+| 
+| Apache Curator
+| Copyright The Apache Software Foundation
+| 
+| Apache Hadoop
+| Copyright The Apache Software Foundation
+| 
+| Apache HTTP Components
+| Copyright The Apache Software Foundation
+| 
+| Apache Iceberg
+| Copyright The Apache Software Foundation
+| 
+| Apache Kerby
+| Copyright The Apache Software Foundation
+| 
+| This product includes additional software licensed under the terms
+| of the following licenses, see LICENSE file:
+| * Apache Software License, Version 2.0
+| * Creative Commons 1.0 Universal
+| * Eclipse Distribution License, Version 1.0
+| * Eclipse Public License, Version 1.0
+| * Eclipse Public License, Version 2.0
+| * GNU General Public License, Version 2 with the GNU Classpath Exception
+| 
```

Comment:
NB: I checked on Nessie "license report" (https://github.com/projectnessie/nessie/releases/download/nessie-0.102.2/nessie-aggregated-license-report-0.102.2.zip). It looks OK for me because:
1. It's GPL with GNU Classpath Exception
2. It's dual licenses (GPL + CDDL or GPL + EPL)

Reply:
GPL, even with the classpath exception, is a Category X license: https://www.apache.org/legal/resolved.html

We can't include it in a release. The Nessie docs need to be updated to remove this before we can ship a release with Nessie.

Reply:
The Nessie team informed me that they checked and cleanup their `NOTICE` (no actual GPL code copied): https://github.com/projectnessie/nessie/blob/main/NOTICE

They also checked the binary distribution and no GPL dependency bundled.

They said they gonna do a new release. I'm already updating here.

Reply:
Thanks @jbonofre for checking with them!

Reply:
We updated to Nessie 0.102.4 that fixed their `NOTICE` in regards of GPL.

Reply:
I think that this is still too unclear to include in Iceberg. The [Nessie binary NOTICE](https://github.com/projectnessie/nessie/blob/main/NOTICE-BINARY-DIST) file that applies to Jars (binaries) still includes mention of GPL code. The fixed Nessie NOTICE states that it applies to [Nessie source code](https://github.com/projectnessie/nessie/blob/main/NOTICE#L1) so this is still not clear from the documentation.

Reply:
I've updated the Nessie NOTICE with the 0.102.5 release.